### PR TITLE
Flickering images fix

### LINKED ISSUE #74: flickering images
prevent the flickering by checking if the new item is the same as the previous item

### PR DESCRIPTION
**Pull request by akrol95**
- Flickering images were mainly caused by too many reload operations and changing last-changed timestamp every time when a document with cover url from the API was retrieved.

DocumentVisualEqualityChecker has been introduced. It's a mechanism checking if elements on the list are visually the same as in old list, and if so, there is no reload. It correctly handle things like changing content language, app language etc.
Initially I placed this logic in individual document classes to compare with other ones (so e.g. in Album.cs, Contributor.cs), but as this classes are in API project of our solution, it's layer mixing and I abandoned this idea
The best idea would be to create Presentation Objects in Core project, and place that logic there in every file, but for now it's too much work.

The problematic thing now is when someone adds new property in some document, which can also have influence to refreshing from UI perspective, it will also be needed to update the logic in DocumentVisualEqualityChecker. 

- Some logic has been introduced to reduce refreshing of the views, when navigating through Bottom Bar.
- Cache for browse section has been introduced (so the load time is much faster now).

Closes #74

### CODE REVIEW

**Comment by kkuepper** [comment]
discussed two things:
- should try using the disk cache
- covers that are outside of view take long to load. we should probably pre-load them.

File: BMM.Core/Constants/ImageServiceConstants.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+using System;
+
+namespace BMM.Core.Constants
+{
+    public class ImageServiceConstants
+    {
+        public const string ImageCacheFolder = "img";
+        public const int ImageCacheMemorySize = 100 * 1024 * 1024; //100 MB
+
+        public static readonly TimeSpan DiskCacheDuration = TimeSpan.FromDays(7);
```
Comment (kkuepper) [inline BMM.Core/Constants/ImageServiceConstants.cs:10]: why not 30 days?

Comment (kkuepper) [reply BMM.Core/Constants/ImageServiceConstants.cs:10]: the url changes when a new version is available. that should allow us to cache it for a long time

Comment (akrol95) [reply BMM.Core/Constants/ImageServiceConstants.cs:10]: Changed to 30 days

**Review by kkuepper — APPROVED** [review]

**Pull request merged**